### DISCUSSION TITLE
Custom Modules & Customizations

### DISCUSSION THREAD

**Discussion by zilmehta-ets** [saleor/saleor#12820] [Q&A]
Hi, 

I'm currently exploring Saleor as an eCommerce Marketplace for our store. 

I needed an understanding on how the custom modules should be approached. 
For example, for my store along with the key eCommerce features, if I want to add a custom module let's say "Book an Appointment", so what should be the approach to have that module as a part of Saleor? 

Do I need to develop a Python plugin that can be integrated as a third-party app in the Saleor admin? 
What happens to the plugin in case there are any updates in the Saleor version?

Secondly, we'll be having a distributed architecture where in the product related data in Saleor will be coming through SAP ECC6 Inventory Management System. So the question is, if at all product enrichment is needed before displaying the products in frontend, what should be the approach? A PIM tool is needed between SAP and Saleor? If yes, which PIM is recommended? If no, can the product enrichment be done within Saleor?

**Comment by timuric** [comment]
Hi @zilmehta-ets we recommend to take microservice approach and extend Saleor using GraphQL and webhooks. 
If you need to extend functionality of the dashboard you could use apps.
